### PR TITLE
Add troubleshooting docs

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

**Comment by incendial** [comment]
@GroovinChip Hi! May I ask you to review the troubleshooting docs? Is it accurate from your standpoint?

**Comment by GroovinChip** [comment]
> @GroovinChip Hi! May I ask you to review the troubleshooting docs? Is it accurate from your standpoint?

Sure, I'd be happy to!

**Comment by incendial** [comment]
@dkrutskikh should we actually add a step about clearing the plugin cache (`.plugin_manager` folder)?

File: TROUBLESHOOTING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+If the plugin is not working as you'd expect it to work, please consider going through the following steps before creating [an issue](https://github.com/dart-code-checker/dart-code-metrics/issues/new?assignees=dkrutskikh&labels=bug&template=bug-report.md&title=%5BBUG%5D+):
+
+1. Check that the plugin is added to an `analyzer` entry in the `analysis_options.yaml` as described in the [Configuration](./README#Configuration) section.
+
+2. Check that the `dart_code_metrics` entry in the `analysis_options.yaml` is configured correctly. Note, that you need to add each rule or metric you want to be checked to the config and there is **no** default rule or metric lists config.
+
+3. Check that the `dart_code_metrics` package is added as a dev dependency to the same package, where the plugin entry added to the `analysis_options.yaml`. If you use a separate sub package for a analyzer configuration, please follow [this discussion](https://github.com/dart-code-checker/dart-code-metrics/issues/254).
```
Comment (GroovinChip) [inline TROUBLESHOOTING.md:11]: > If you use a separate sub package for a analyzer configuration
Change to 
> If you use a separate sub package for an analyzer configuration

**Review by GroovinChip — CHANGES_REQUESTED** [review]
I think it would be worth adding a note that restarting the analyzer and/or restarting the IDE (and invalidating caches for Android Studio) could help sometimes.

**Review by GroovinChip — CHANGES_REQUESTED** [review]
I think it's also important to note that configurable rules require 4 spaces instead of 2:
```yml
- member-ordering:
1234alphabetize: true
```
This tripped me up when I was configuring it myself.

File: TROUBLESHOOTING.md (hunk 1 above)
Comment (incendial) [reply TROUBLESHOOTING.md]: Done.

**Comment by incendial** [comment]
> I think it would be worth adding a note that restarting the analyzer and/or restarting the IDE (and invalidating caches for Android Studio) could help sometimes.

Good point, I've added it as a step 4.

> I think it's also important to note that configurable rules require 4 spaces instead of 2:

Yeah, this is reasonable too.

**Review by GroovinChip — APPROVED** [review]

**Review by GroovinChip — APPROVED** [review]

**Review by dkrutskikh — APPROVED** [review]

**Comment by incendial** [comment]
@GroovinChip thank you for the review!

**Comment by GroovinChip** [comment]
> @GroovinChip thank you for the review!

You're very welcome 😁

**Pull request merged**